src/main/java/com/ibm/watson/modelmesh/Metrics.java

```diff
@@ -220,10 +237,15 @@ public PrometheusMetrics(Map<String, String> params, Map<String, String> infoMet
                 }
 
                 if (m == API_REQUEST_TIME || m == API_REQUEST_COUNT || m == INVOKE_MODEL_TIME
-                    || m == INVOKE_MODEL_COUNT || m == REQUEST_PAYLOAD_SIZE || m == RESPONSE_PAYLOAD_SIZE) {
-                    builder.labelNames("method", "code");
+                        || m == INVOKE_MODEL_COUNT || m == REQUEST_PAYLOAD_SIZE || m == RESPONSE_PAYLOAD_SIZE) {
+                    if (this.enablePerModelMetrics && m.type != COUNTER_WITH_HISTO) {
+                        builder.labelNames("method", "code", "modelId");
+                    } else {
+                        builder.labelNames("method", "code");
+                    }
+                } else if (this.enablePerModelMetrics && m.type != GAUGE && m.type != COUNTER && m.type != COUNTER_WITH_HISTO) {
+                    builder.labelNames("modelId");
```

Comment:
I think we should add the `modelId` label based on a specific list of metric names rather than the metric types. Since in general there's no reason why a particular type should be model-specific or not.

Reply:
we are adding the label on a list of metrics 
```
if (m == API_REQUEST_TIME || m == API_REQUEST_COUNT || m == INVOKE_MODEL_TIME || m == INVOKE_MODEL_COUNT || m == REQUEST_PAYLOAD_SIZE || m == RESPONSE_PAYLOAD_SIZE)
```
I removed the check for `m.type != COUNTER_WITH_HISTO`